authoringAssistant/src/main/java/explorableviz/transparenttext/QueryContext.java

```diff
@@ -220,4 +256,39 @@ public static ArrayList<QueryContext> loadCases(String casesFolder, int numInsta
                 });
         return queryContexts;
     }
+
+    public void replaceVariables(Map<String, String> variables, Random random) {
+        String code = this.getCode();
+        String expected = this.getExpected();
+
+        for (Map.Entry<String, String> entry : variables.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            String variablePlaceholder = STR."$\{key}$";
+            String replacement = switch (value) {
+                case "RANDOM_INT" -> String.valueOf(random.nextInt(10));
+                case "RANDOM_FLOAT" -> String.format("%.6f", random.nextDouble() * 10);
+                case "RANDOM_STRING" -> getRandomString(8, random).toLowerCase();
+                default -> value;
+            };
+            code = code.replace(variablePlaceholder, replacement);
+            expected = expected.replace(variablePlaceholder, replacement);
+            this.getParagraph().forEach(t -> {
+                t.setValue(t.getValue().replace(variablePlaceholder, replacement));
+            });
+            this.get_loadedDatasets().replaceAll((k, v) -> v.replace(variablePlaceholder, replacement));
```

Comment:
It would be good to be consistent about whether you refer to fields via `this.simpleName` or `this.getSimpleName()`.

Reply:
I refactored the code using in general getters instead of property name